### PR TITLE
fix: Remove Server header and allow all on port 80

### PR DESCRIPTION
**Pull request by sharat87**
Another attempt at #29550, which was reverted. Fallback is not happening if cert provisioning fails _despite_ having the correct header. But with the changes in this PR, since we'll listen on `:80`, fallback _will_ happen when cert provisioning fails due to incorrect domain configuration.

We're also adding [Hurl](https://hurl.dev) based tests. They're not run in any CI yet. That'll come in soon.

## Summary by CodeRabbit

- **Refactor**
	- Improved handling of custom domain configurations in the server setup.
	- Enhanced HTTP to HTTPS redirection for custom domains.

- **Chores**
	- Updated environment variable naming for clarity in custom domain setup.

### CODE REVIEW

File: deploy/docker/fs/opt/appsmith/caddy-reconfigure.mjs [hunk 1]
```diff
@@ -3,25 +3,28 @@ import {dirname} from "path"
 import {spawnSync} from "child_process"
 import {X509Certificate} from "crypto"
 
-const APPSMITH_CUSTOM_DOMAIN = process.env.APPSMITH_CUSTOM_DOMAIN ?? null
+// The custom domain is expected to only have the domain. So if it has a protocol, we ignore the whole value.
+// This was the effective behaviour before Caddy.
+const CUSTOM_DOMAIN = (process.env.APPSMITH_CUSTOM_DOMAIN || "").replace(/^https?:\/\/.+$/, "")
```
[the remark this answers is not in the document]
Comment (sharat87) [reply deploy/docker/fs/opt/appsmith/caddy-reconfigure.mjs:8]: Not accurate. We're not deleting that env variable.

File: deploy/docker/fs/opt/appsmith/caddy-reconfigure.mjs [hunk 2]
```diff
@@ -3,25 +3,28 @@
...
     try {
       fs.accessSync(fullChainPath, fs.constants.R_OK)
       console.log("Old Let's Encrypt cert file exists, now checking if it's expired.")
       if (!isCertExpired(fullChainPath)) {
         certLocation = letsEncryptCertLocation
       }
-    } catch (_) {
+    } catch {
       // no certs there either, ignore.
     }
```
[the remark this answers is not in the document]
Comment (sharat87) [reply deploy/docker/fs/opt/appsmith/caddy-reconfigure.mjs:29]: Not necessary. No matter the error, we want to go ahead with no explicit TLS configuration.

File: deploy/docker/route-tests/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    --resolve local.com:80:127.0.0.1 \
+    --resolve custom-domain.com:80:127.0.0.1 \
+    --resolve custom-domain.com:443:127.0.0.1 \
+    "$@"
+}
+
+#if [[ "${OPEN_SHELL-}" == 1 ]]; then
+  # Open shell for debugging after this script is done.
+  trap bash EXIT
+#fi
```
[the remark this answers is not in the document]
Comment (sharat87) [reply deploy/docker/route-tests/entrypoint.sh]: Thanks.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**